clients/src/main/java/org/apache/kafka/clients/producer/internals/ProduceRequestResult.java

```diff
@@ -32,21 +34,31 @@
     private final CountDownLatch latch = new CountDownLatch(1);
     private volatile TopicPartition topicPartition;
     private volatile long baseOffset = -1L;
+    private volatile long responseTimestamp = Record.NO_TIMESTAMP;
```

Comment:
Mentioned offline, but it would be good to understand the need for these fields to be `volatile`. It seems that the use of the latch is sufficient to ensure visibility from the memory model perspective.